BenchPress/Generators.Tests/TestGeneratorTests.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Generators.LanguageProviders;
+
+namespace Generators.Tests
+{
+  [TestClass]
+  public class TestGeneratorTests
+  {
+    [TestMethod]
+    public void Should_Throw_If_Unknown_TestType()
+    {
+      // arrange
+      var resourceType = "Microsoft.Insights/actionGroups";
+      var resourceName = "Name";
+      var extraProperties = new Dictionary<string, string>();
+
+      var metadata = new TestMetadata(resourceType, resourceName, extraProperties);
+      var definition = new TestDefinition(metadata, (TestType)1);
```

Comment:
So, this might be a bit brittle. If a TestType is added, this test will no longer pass and the test will need to be updated. Maybe use `Int32.MaxValue` or something to that effect.